libdbus-sys/build.rs

```diff
@@ -1,8 +1,320 @@
 extern crate pkg_config;
+use std::error::Error;
+
+// Constants and gated use statements
+#[cfg(feature = "vendored")]
+mod consts {
+    pub use std::collections::HashMap;
+    pub use std::env;
+    pub use std::fs::File;
+    pub use std::io::Write;
+
+    // Version of vendored libdbus
+    pub const DBUS_VERSION: &str = "\"1.15.2\"";
+    pub const DBUS_VERSION_MAJOR: u32 = 1;
+    pub const DBUS_VERSION_MINOR: u32 = 15;
+    pub const DBUS_VERSION_MICRO: u32 = 2;
+
+    // Location of dbus-daemon socket
+    pub const DBUS_SYSTEM_BUS_DEFAULT_ADDRESS: &str = "/run/dbus/system_bus_socket";
+    pub const DBUS_SESSION_SOCKET_DIR: &str = "/tmp";
+}
+
+#[cfg(feature = "vendored")]
+use consts::*;
+
+/// Source Files Required
+#[cfg(feature = "vendored")]
+const SOURCE_FILES: &'static [&'static str] = &[
+    "./vendor/dbus/dbus/dbus-address.c",
+    "./vendor/dbus/dbus/dbus-auth.c",
+    "./vendor/dbus/dbus/dbus-bus.c",
```

Comment:
A lot of code in this file might be copy-pasted and slightly changed from `Makefile.am` and `configure.ac`. I'm a bit worried that it makes the code a bit fragile, e g when libdbus updates and adds another source code file the vendoring will break. 

OTOH, trying to autogenerate things might be overkill...or what do you think?

Reply:
Yeah it was taken directly from their [Meson config](https://github.com/freedesktop/dbus/blob/master/dbus/meson.build#L31).

I think combined with pinning to a specific libdbus release (discussed in the other comment) it shouldn't break user's builds. 

So `libdbus-sys` 0.2.3 will be pinned to `libdbus` 1.15.2. And upgrading to `libdbus` 1.15.3 would require updating and testing a new `libdbus-sys` release. So broken builds should be caught during testing before we roll out another version of `libdbus-sys`.

The benefits of using the `cc` crate directly instead of calling `configure`/`cmake`/`meson` is that it doesn't swap the `libdbus-dev-1` dependency for other build-time dependencies. And allows downstream crates/users to simply use cargo to build.